crates/valence_command_derive/README.md

```diff
@@ -0,0 +1,104 @@
+Simplify the creation of Valence commands with a derive macro.
+
+## Usage
+
+```rust
+#[derive(Command, Debug, Clone)]
+#[paths("teleport", "tp")]
+#[scopes("valence:command:teleport")]
+enum Teleport {
+    #[paths = "{location}"]
+    ExecutorToLocation { location: Vec3Parser },
+    #[paths = "{target}"]
+    ExecutorToTarget { target: EntitySelector },
+    #[paths = "{from} {to}"]
+    TargetToTarget {
+        from: EntitySelector,
+        to: EntitySelector,
+    },
+    #[paths = "{target} {location}"]
+    TargetToLocation {
+        target: EntitySelector,
+        location: Vec3Parser,
+    },
+}
+
+#[derive(Command, Debug, Clone)]
+#[paths("gamemode", "gm")]
+#[scopes("valence:command:gamemode")]
+enum Gamemode {
+    #[paths("survival", "{/} gms")]
+    Survival,
+    #[paths("creative", "{/} gmc")]
+    Creative,
+    #[paths("adventure", "{/} gma")]
+    Adventure,
+    #[paths("spectator", "{/} gmsp")]
+    Spectator,
+}
+
+#[derive(Command, Debug, Clone)]
+#[paths("test", "t")]
+#[scopes("valence:command:test")]
+#[allow(dead_code)]
+enum Test {
+    // 3 literals with an arg each
+    #[paths("a {a} b {b} c {c}", "{a} {b} {c}")]
+    A { a: String, b: i32, c: f32 },
+    // 2 literals with an arg last being optional (Because of the greedy string before the end
+    // this is technically unreachable)
+    #[paths = "a {a} {b} b {c?}"]
+    B {
+        a: Vec3Parser,
+        b: GreedyString,
+        c: Option<String>,
+    },
+    // greedy string optional arg
+    #[paths = "a {a} b {b?}"]
+    C { a: String, b: Option<GreedyString> },
+    // greedy string required arg
+    #[paths = "a {a} b {b}"]
+    D { a: String, b: GreedyString },
+    // five optional args and an ending greedyString
+    #[paths("options {a?} {b?} {c?} {d?} {e?}", "options {b?} {a?} {d?} {c?} {e?}")]
+    E {
+        a: Option<i32>,
+        b: Option<QuotableString>,
+        c: Option<Vec2Parser>,
+        d: Option<Vec3Parser>,
+        e: Option<GreedyString>,
+    },
+}
+```
+
+## attributes
+
+### `#[paths(...)]` or `#[paths = "..."]`
+
+The `#[paths(...)]` or `#[paths = "..."]` attribute is used to specify the different paths that can be used to invoke
+the command. The paths are specified as string literals, where any arguments are enclosed in curly braces `{}`.
+The arguments are then mapped to fields in the command enum variant.
```

Comment:
If we are going to keep the name paths though, this needs to do a better job at explaining that Minecraft commands are parsed according to a graph structure. This is could be the first thing that someone looks at if they want to implement commands, and the context would be helpful for understanding the API.

Perhaps a mermaid diagram for the teleport command would be a useful visual.